driver/src/main/java/org/kaazing/k3po/driver/internal/control/handler/ControlServerHandler.java

```diff
@@ -106,6 +109,28 @@ public void prepareReceived(final ChannelHandlerContext ctx, MessageEvent evt) t
             return;
         }
 
+        if ((lastRobot != null && ! lastRobot.getDisposedFuture().isDone()) || ! testFinished.compareAndSet(true, false)) {
```

Comment:
You can probably create a queue by just chaining disposedFutures and have it very similar to what you have now: something like 
```
 public synchronized void schedulePrepare(){
        Robot robot = new Robot();  // I think this is already done outside the method
        if(queueTail == null){
            startNow();
        }else{
            queueTail.get().addListener(new ChannelFutureListener() {
                @Override
                public void operationComplete(ChannelFuture future) throws Exception {
                    startNow();
                }
            });
        }
        queueTail.set(robot.getDisposedFuture());
    }
```